File: services/security/backend/files/app/models/wazuh.py
from typing import List, Optional
from pydantic import BaseModel


class WazuhAgent(BaseModel):
    id: str
    ip: str
    name: str
    status: str


class WazuhSCAPolicy(BaseModel):
    failed: int
    passed: int
    unapplicable: int
    name: str
    id: str


class WazuhSCAPolicyCheck(BaseModel):
    description: str
    directory: Optional[str]
    file: Optional[str]
    command: Optional[str]
    rationale: str
    references: Optional[str]
    remediation: Optional[str]
    result: str
    title: str


class WazuhAgentVulnerability(BaseModel):
    severity: str
    # updated: str;
    version: str
    type: str
    name: str
    external_references: List[str]
    condition: str
    detection_time: str
    cvss3_score: float
    cvss2_score: float
    published: str
    architecture: str
    cve: str
    status: str
    title: str


class FileIntegrityAlertRule(BaseModel):
    level: int
    description: str


class WazuhAgentFileIntegrityAlert(BaseModel):
    id: str
    path: str
    rule: FileIntegrityAlertRule
    full_log: str

File: services/security/frontend/files/src/types/wazuh.ts
export interface IAgentInformation {
  id: string;
  ip: string;
  name: string;
  status: string;
}

export interface IAgentSCAPolicy {
  failed: number;
  passed: number;
  unapplicable: number;
  name: string;
  id: string;
}

export interface  IAgentSCAPolicyCheck {
  description: string;
  directory?: string;
  file?: string;
  command?: string;
  rationale: string;
  references?: string;
  remediation?: string;
  result: string;
  title: string;
}

export interface IAgentVulnerability {
  severity: string;
  // updated: string;
  version: string;
  type: string;
  name: string;
  external_references: string[];
  condition: string;
  detection_time: string;
  cvss3_score: number;
  cvss2_score: number;
  published: string;
  architecture: string;
  cve: string;
  status: string;
  title: string;
}

export interface IAgentFileIntegrityAlert {
  id: string;
  path: string;
  rule: {
    level: number;
    description: string;
  };
  full_log: string;
}

export function agentStatusGate(agent: IAgentInformation, fn: CallableFunction) {
  // non-active agents can not check policies
  if (agent.status === "active") {
    fn();
  }
}